src/Analysis/Ast/Impl/Extensions/PythonTypeExtensions.cs

```diff
@@ -28,18 +28,14 @@ public static bool IsGenericParameter(this IPythonType value)
         public static bool IsGeneric(this IPythonType value)
             => value is IGenericTypeDefinition || value is IGenericType || (value is IPythonClassType c && c.IsGeneric());
 
-        public static void TransferDocumentationAndLocation(this IPythonType src, IPythonType dst) {
-            if (src != null && dst is PythonType pt) {
-                pt.TrySetTypeId(dst.TypeId);
+        public static void TransferDocumentationAndLocation(this IPythonType s, IPythonType d) {
+            if (s != null && s != d && s is PythonType src && d is PythonType dst) {
```

Comment:
I think we don't need `s != null` check here. Pattern matching will fail for `s == null`